ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/parallelization/ParallelizedConcurrentTransactionProcessor.java

```diff
@@ -0,0 +1,268 @@
+       */
+      CompletableFuture.runAsync(
+          () ->
+              runTransaction(
+                  worldState,
+                  blockHeader,
+                  transactionLocation,
+                  transaction,
+                  miningBeneficiary,
+                  blockHashLookup,
+                  blobGasPrice,
+                  privateMetadataUpdater),
+          executor);
+    }
+  }
+
+  @VisibleForTesting
+  public void runTransaction(
+      final MutableWorldState worldState,
+      final BlockHeader blockHeader,
+      final int transactionLocation,
+      final Transaction transaction,
+      final Address miningBeneficiary,
+      final BlockHashOperation.BlockHashLookup blockHashLookup,
+      final Wei blobGasPrice,
+      final PrivateMetadataUpdater privateMetadataUpdater) {
+    try (final DiffBasedWorldState roundWorldState =
+        new BonsaiWorldState(
+            (BonsaiWorldState) worldState, new NoopBonsaiCachedMerkleTrieLoader())) {
+      roundWorldState.freeze(); // make the clone frozen
+      final ParallelizedTransactionContext.Builder contextBuilder =
+          new ParallelizedTransactionContext.Builder();
+      final DiffBasedWorldStateUpdateAccumulator<?> roundWorldStateUpdater =
+          (DiffBasedWorldStateUpdateAccumulator<?>) roundWorldState.updater();
+      final TransactionProcessingResult result =
+          transactionProcessor.processTransaction(
+              roundWorldStateUpdater,
+              blockHeader,
+              transaction,
+              miningBeneficiary,
+              new OperationTracer() {
+                @Override
+                public void traceBeforeRewardTransaction(
+                    final WorldView worldView,
+                    final org.hyperledger.besu.datatypes.Transaction tx,
+                    final Wei miningReward) {
+                  /*
+                   * This part checks if the mining beneficiary's account was accessed before increasing its balance for rewards.
+                   * Indeed, if the transaction has interacted with the address to read or modify it,
+                   * it means that the value is necessary for the proper execution of the transaction and will therefore be considered in collision detection.
+                   * If this is not the case, we can ignore this address during conflict detection.
+                   */
+                  if (transactionCollisionDetector
+                      .getAddressesTouchedByTransaction(
+                          transaction, Optional.of(roundWorldStateUpdater))
+                      .contains(miningBeneficiary)) {
+                    contextBuilder.isMiningBeneficiaryTouchedPreRewardByTransaction(true);
+                  }
+                  contextBuilder.miningBeneficiaryReward(miningReward);
```

Comment:
Running gradle check with code coverage doesn't show this covered.  Definitely would be nice to have this covered in test